File: projects/order-creation-queries/app/src/main/java/com/robinfood/app/usecases/getordersdaily/IGetOrdersDailyUseCase.java
package com.robinfood.app.usecases.getordersdaily;

import com.robinfood.core.dtos.OrderDailyDTO;
import com.robinfood.core.enums.Result;

import java.util.List;

/**
 * Use case that returns the orders of the day.
 */
public interface IGetOrdersDailyUseCase {

    /**
     * Get orders of the day depending on the time zone.
     *
     * @param storeId Store identifier
     * @param timeZone Client time zone
     * @return Daily Order List ready to invoice
     */
    Result<List<OrderDailyDTO>> invoke(Long storeId, String timeZone);
}

File: projects/posv2/app/src/main/kotlin/com/robinfood/app/usecases/orderdaily/IGetOrdersDailyUseCase.kt
package com.robinfood.app.usecases.orderdaily

import com.robinfood.core.dtos.OrderDailyResponseDTO
import com.robinfood.core.enums.Result

/**
 * Use case that retrieves all orders to be invoiced
 */
interface IGetOrdersDailyUseCase {

    suspend fun invoke(
            token: String,
            timeZone: String,
            storeId: Long
    ): Result<OrderDailyResponseDTO>
}